src/SSHDebugPS/Docker/DockerHelper.cs

```diff
@@ -10,13 +10,111 @@
 using Microsoft.SSHDebugPS.SSH;
 using Microsoft.SSHDebugPS.Utilities;
 
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+
 namespace Microsoft.SSHDebugPS.Docker
 {
     public class DockerHelper
     {
         private const string dockerPSCommand = "ps";
         // --no-trunc avoids parameter truncation
         private const string dockerPSArgs = "-f status=running --no-trunc --format \"{{json .}}\"";
+
+        public static string GetDockerOutputString(string hostname, string dockerCommand, string dockerArgs)
+        {
+            string path = null;
+            try
+            {
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    path = GetDockerOutputStringImpl(hostname, dockerCommand, dockerArgs);
+                });
+            }
+            catch (Exception)
+            {
+            }
+            return path;
+        }
+
+        public static string GetDockerOutputStringImpl(string hostname, string dockerCommand, string dockerArgs)
```

Comment:
If you are going to make an Impl method to back the public method, mark it `private`. If your intention is to be able to call either from the public surface, then don't create the impl method.

Reply:
Call this GetOutputFromDockerCommand or something and reuse it for the other calls in this file.